hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncScanSingleRegionRpcRetryingCaller.java

```diff
@@ -418,19 +416,14 @@ private void onError(Throwable error) {
       completeExceptionally(!scannerClosed);
       return;
     }
-    long delayNs;
-    long pauseNsToUse =
-      HBaseServerException.isServerOverloaded(error) ? pauseNsForServerOverloaded : pauseNs;
-    if (scanTimeoutNs > 0) {
-      long maxDelayNs = remainingTimeNs() - SLEEP_DELTA_NS;
-      if (maxDelayNs <= 0) {
-        completeExceptionally(!scannerClosed);
-        return;
-      }
-      delayNs = Math.min(maxDelayNs, getPauseTime(pauseNsToUse, tries - 1));
-    } else {
-      delayNs = getPauseTime(pauseNsToUse, tries - 1);
+
+    OptionalLong maybePauseNsToUse = pauseManager.getPauseNsFromException(error,
+      remainingTimeNs() - SLEEP_DELTA_NS, tries, scanTimeoutNs > 0);
```

Comment:
Passing in `scanTimeoutNs > 0` along with the `remainingTimeNs()` seems a bit awkward/redundant. 

We are already constructing our PauseManager in the caller construction, and passing in the various pauseNs. Should we also pass in the correct operationTimeNs/scanTimeNs in the constructor? Then we can move `remainingTimeNs()` into pause manager and simplify the arguments here, along with the awkward Boolean?

Reply:
👍 good idea, I just pushed this change